src/main/java/reactor/netty/resources/NewConnectionProvider.java

```diff
@@ -81,6 +86,41 @@
 		});
 	}
 
+	@Override
+	@SuppressWarnings("unchecked")
+	public <CONF extends TransportConfig, CONN extends Connection> Mono<CONN> acquire(TransportClientConfig<CONF, CONN> config) {
+		return Mono.create(sink -> {
```

Comment:
do we need a `create`? (ie. would `defer` work?)

Reply:
This method returns `Mono<CONN>` where `CONN` extends `Connection` while `reactor.netty.transport.TransportConnector#connect` returns `Mono<Channel>`

The special `ConnectionObserver` will provide the `Connection` when it is available